code/modules/ftl/code/game/machinery/ftl_shieldgen.dm

```diff
@@ -167,22 +170,17 @@
 		S.in_dir = dirs[2]
 		shield_barrier_objs += S
 
-/obj/machinery/ftl_shieldgen/proc/drop_physical(delayed = 0)
+/obj/machinery/ftl_shieldgen/proc/drop_physical()
 	if(!do_update)
 		return
-	if(delayed)
-		do_update = 0
-		while(shield_barrier_objs.len)
-			var/obj/to_remove = pick(shield_barrier_objs)
-			qdel(to_remove)
-			shield_barrier_objs -= to_remove
-			if(prob(10))
-				sleep(1)
-		do_update = 1
-	else
-		for(var/obj/effect/ftl_shield/S in shield_barrier_objs)
-			qdel(S)
-		shield_barrier_objs.Cut()
+	do_update = FALSE
+	while(shield_barrier_objs.len)
+		var/obj/to_remove = pick(shield_barrier_objs)
+		qdel(to_remove)
+		shield_barrier_objs -= to_remove
+		if(prob(10))
+			sleep(1)
+	do_update = TRUE
```

Comment:
instead of using extra vars making sure this isn't called too much, make all calls to this proc a unique addtimer

Reply:
how would that work?